tests/testsuite/precise_pre_release.rs

```diff
@@ -76,44 +76,92 @@ fn update_pre_release() {
 }
 
 #[cargo_test]
-fn update_pre_release_differ() {
+fn update_pre_release_range() {
```

Comment:
How much of these test changes can happen in a commit before the fix?  Separating out expanded coverage from behavior changes would make this a lot easier to review

Reply:
I split the change into the first commit, which include the two cases, 

- `pre_release_should_unmatched` explains the drawback that needs to be fixed in propose semantic
- `pre_release_should_matched` can be treat as a normal test that must be meet